src/config/parser.rs

```diff
@@ -148,6 +161,7 @@ impl Config {
             safe_search,
             #[cfg(any(feature = "redis-cache", feature = "memory-cache"))]
             cache_expiry_time,
+            proxy,
```

Comment:
```suggestion
            #[cfg(feature = "socks")]
            proxy,
```

I would suggest having this behind a `cfg` feature macro as this will only add the following code to the binary if the `socks` feature has been enabled during build otherwise it will not include it thus reducing binary size and removing unneeded code. 🙂